src/Peering/Peering.Test/ScenarioTests/CreateNewDirectConnectionTests.cs

```diff
@@ -0,0 +1,125 @@
+        {
+            this.logger = new ServiceManagement.Common.Models.XunitTracingInterceptor(output);
+            ServiceManagement.Common.Models.XunitTracingInterceptor.AddToContext(this.logger);
+        }
+
+        /// <summary>
+        /// The test new direct connection high bandwidth.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestNewDirectConnectionHighBandwidth()
+        {
+            TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewDirectConnectionHighBandwidth");
+        }
+
+        /// <summary>
+        /// The test new direct connection low bandwidth.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestNewDirectConnectionLowBandwidth()
+        {
+            TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewDirectConnectionLowBandwidth");
+        }
+
+        /// <summary>
+        /// The test new direct connection no session.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestNewDirectConnectionNoSession()
+        {
+            TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewDirectConnectionNoSession");
+        }
+
+        /// <summary>
+        /// The test new direct connection with v 4.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestNewDirectConnectionWithV4()
+        {
+            TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewDirectConnectionWithV4");
+        }
+
+        /// <summary>
+        /// The test new direct connection with v 4 v 6.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestNewDirectConnectionWithV4V6()
+        {
+            TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewDirectConnectionWithV4V6");
+        }
+
+        /// <summary>
+        /// The test new direct connection with v 6.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestNewDirectConnectionWithV6()
+        {
+            TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewDirectConnectionWithV6");
+        }
+
+        /// <summary>
+        /// The test new direct connection wrong v 4.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.CheckIn)]
+        public void TestNewDirectConnectionWrongV4()
+        {
+            TestController.NewInstance.RunPowerShellTest(this.logger, "Test-NewDirectConnectionWrongV4");
+        }
+
+        /// <summary>
+        /// The test new direct connection wrong v 6.
+        /// </summary>
+        [Fact]
+        [Trait(Category.AcceptanceType, Category.DesktopOnly)]
```

Comment:
Yes this is an in memory call and there is an issue with the `r`n when running this test from the cloud build. I fixed the issue by doing a match rather than Equals.